### PR TITLE
QMCA save averaged data to scalar files

### PR DESCRIPTION
**Pull request by camelto2**
This enables qmca to save the averaged data from the -a option to scalar files for
each series. This will allow one to use qmc-fit to perform a time-step
extrapolation on twist averaged data, for example. One only needs to add the option --save_average, and it will store avg.sXXX.scalar.dat files for each series. Then use qmc-fit on the new scalar files to extrapolate your twisted averaged data.

### CODE REVIEW

**Comment by markdewing** [comment]
okay to test

File: nexus/bin/qmca [hunk 1]
```diff
@@ -2677,6 +2683,18 @@ QMCA examples:
             #end for
         #end for
         opt = self.options
+        if opt.average and opt.save_average:
+            for fmap in self.file_map:
+                ndata = len(fmap.data['LocalEnergy'])
```
Comment (jtkrogel) [inline nexus/bin/qmca]: Use fmap.data.first() instead.  Depending on the method, LocalEnergy may or may not be the key used for the total energy.

File: nexus/bin/qmca [hunk 2]
```diff
@@ -2677,6 +2683,18 @@ QMCA examples:
...
+                data = [arange(ndata)]
+                keys = 'index'
+                fmt = ['%i']
+                for key in fmap.data.keys():
+                    data.append(fmap.data[key])
+                    keys+='     {}'.format(key)
+                    fmt.append('%.10e')
```
Comment (jtkrogel) [inline nexus/bin/qmca]: Format control expressions to match surrounding code, i.e. use "#end if" and "#end for".

**Comment by jtkrogel** [comment]
@camelto2 just a couple of small changes, see above.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
@jtkrogel Good to merge?

**Comment by jtkrogel** [comment]
Yes.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
Test this please.

**Pull request merged**